Movie Search/SearchController.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  SearchController.swift
+//  Movie Search
+//
+//  Created by Alexey Grabik on 15/04/2020.
+//
+
+import UIKit
+
+enum SearchStatus {
+    case error(String), nothingRequested, noResult, success
+}
+
+class SearchController: NSObject {
+    
+    private let networkService: NetworkService
+    
+    private var moviesModels = [MovieModel]() {
+        didSet {
+            downloadQueue.cancelAllOperations()
+        }
+    }
+    private let downloadQueue = OperationQueue()
+    
+    var searchPerformedCallback: ((SearchStatus) -> Void)?
+    
+    init(networkService: NetworkService = NetworkAPI()) {
+        self.networkService = networkService
+        self.downloadQueue.maxConcurrentOperationCount = Constants.maxConcurrentOperationCount
+        super.init()
+    }
+    
+    func search(for movieTitle: String) {
+        NSObject.cancelPreviousPerformRequests(withTarget: self) 
+        perform(#selector(self.performSearch(_:)),
+                with: movieTitle,
+                afterDelay: Constants.searchThrottleDelay)
+    }
+}
+
+private extension SearchController {
+    
+    class func status(with networkError: NetworkServiceError?) -> SearchStatus? {
+        guard let error = networkError else { return nil }
+        switch error {
+        case .badResponse:
```

Comment:
maybe move this as an extension of `NetworkingServiceError` that is private to this file, then you can get rid of this entire class function